File: python/rna-transcription/rna_transcription.py


dna_to_rna = {
        'G': 'C',
        'C': 'G',
        'A': 'U',
        'T': 'A'
}

def to_rna(dna_strand):
    rna = ""
    for c in dna_strand:
        x = dna_to_rna.get(c)
        if ( x == None ):
            raise ValueError("Wrong input!")
        rna = rna + x
    return rna

File: typescript/rna-transcription/rna-transcription.ts
// Thanks @jaemk

class Transcriptor {
    private conv(s: string): string | undefined {
        switch (s) {
            case 'A': return 'U'
            case 'T': return 'A'
            case 'G': return 'C'
            case 'C': return 'G'
            default: return undefined
        }
    }
    toRna(seq: string) {
        let rna = ''
        for (const c of seq) {
            const piece = this.conv(c)
            if (!piece) { throw('Invalid input DNA.') }
            rna += piece
        }
        return rna
    }
}

export default Transcriptor

